src/Aspire.Hosting.Garnet/GarnetBuilderExtensions.cs

```diff
@@ -83,7 +83,7 @@ public static class GarnetBuilderExtensions
             isReadOnly);
         if (!isReadOnly)
         {
-            builder.WithPersistence();
+            builder.WithPersistence(interval: null);
```

Comment:
This is unfortunate that the overloads are conflicting. Can we solve this somehow? Potentially by making the Obsolete overload not have default parameters?

Reply:
@eerhardt, do you mean something like this?
```csharp
[Obsolete("This method is obsolete and will be removed in a future version. Use the overload without the keysChangedThreshold parameter.")]
  public static IResourceBuilder<GarnetResource> WithPersistence(this IResourceBuilder<GarnetResource> builder,
      TimeSpan? interval, long keysChangedThreshold)
      => WithPersistence(builder, interval);
```

Reply:
Yes, I think removing the optional parameters on the Obsolete overload is a good compromise here.

@mitchdenny - thoughts?

Reply:
Agree with the strategy.